Sources/SourceControl/GitRepository.swift

```diff
@@ -46,18 +46,25 @@ public class GitRepositoryProvider: RepositoryProvider {
         return try result.utf8Output()
     }
 
-    public func fetch(repository: RepositorySpecifier, to path: AbsolutePath) throws {
+    public func fetch(repository: RepositorySpecifier, to path: AbsolutePath, update: Bool) throws {
         // Perform a bare clone.
         //
         // NOTE: We intentionally do not create a shallow clone here; the
         // expected cost of iterative updates on a full clone is less than on a
         // shallow clone.
+        if update {
```

Comment:
Originally I wanted to keep this in line with `cloneCheckout` which also has a boolean flag for an assumption about the repository. Since it is quite simple to check for a pre-existing git repository in this case, I think we can just use `localFileSystem.exists(path.appending(component: ".git"))`.

Reply:
Makes sense.  I didn't notice that there was already a flag in one of the others.  Thanks for simplifying it in the latest iterations; it can always be added back if it's really needed.